tests/test_train_test_split.py

```diff
@@ -0,0 +1,63 @@
+from data.iterators import count_datapoints, create_train_dataset_iterator, create_test_dataset_iterator
```

Comment:
Can we test this on datasets where the number of datapoints % batch_size != 0, where batch_size % 2 != 0, batch_size > number of datapoints, *infinite = True*, labeler out of bounds, invalid batch_size, invalid dataset_path? If you think of other edge cases please add them too.

Reply:
I get all of these but why _infinite = True_? As far as I understand, it just wraps the generator in a while loop (the test would run the same, except I need to break out of the iteration when collecting data points).

Reply:
Also, why test invalid dataset_path? Isn't it the responsibility of the DatasetManager to handle that case?

Reply:
Tested the other things you mentioned (via the main test, or created new ones).

Reply:
For *infinite = True*, you can verify that we actually go through the data a finite amount of times (3N for example). I don't think we could check it's actually infinite.

True about the *dataset_path*, but we should make both components robust against invalid paths. The test should be in both.

Reply:
Added both!